### ISSUE TITLE
Delay when opening tool belt "sometimes"

### ISSUE THREAD

**Issue by MarioSMB** [gigaherz/ToolBelt#142]
Sometimes when I open the toolbelt it shows like this for half a second:  
![2024-01-05_15 51 34](https://github.com/gigaherz/ToolBelt/assets/1998392/cbccddec-2afa-4fb2-aae2-e4fadfbb0d4e)

Other times it shows the tools instantly - the behaviour seems inconsistent.

**Comment by MarioSMB** [comment]
To clarify, it usually looks like this instantly:  
![2024-01-05_15 52 52](https://github.com/gigaherz/ToolBelt/assets/1998392/43c02483-fdab-4189-b984-745afe23515a)

It also shows as a very small ring for half a second after I release the key sometimes.

**Comment by theboo** [comment]
I am able to reproduce this in 1.19.2. and the cause seems to be connection to the server. Back in August I was living in town and had gigabit fiber. I rent a dedicated server from bloom for my family and we have it running enigmatica 9 expert. I never once had this issue back then and I've used this mod extremely often for hundreds of hours.

In August I sold my house and moved to a rural area where I have starlink. Just now started playing again on the same server and this is constantly happening, makes the toolbelt feel horrible to use. I get about 120 down 15 up and my connection to the server is usually between 60 and 70ms, which is not horrible I don't think.

Ars nouveau does not have this issue with any of their radials(including ones that actually store items like the alchemist crown) and they based their code off this mod.

This does not happen in single player 

https://gyazo.com/8ec9ac41d98cc17f7884339e0765b433

**Comment by Silvertide7** [comment]
I notice this in 1.20.1 as well

**Comment by gigaherz** [comment]
Not at all. It's not an enforcement. The radial menu is a Screen, like the options and such. Minecraft disables movement while interacting with screens by default. In newer versions I have additional logic in the screen that allows it to still process movement controls, bypassing the "pause" (lack of processing) that the game applies on the movement.

The screen opens when you press the keybind, and closes when you release or click. It's all in the client except the actual swapping operation. There should be no server-related delay involved in the animations.
